python/lib/scheduler/dmod/scheduler/resources/resource.py

```diff
@@ -288,28 +301,53 @@ def get_resource_enum_value(cls, enum_type: Union[Type[ResourceAvailability], Ty
                 return val
         return None
 
-    def __eq__(self, other):
+
+    def __eq__(self, other: object):
         if not isinstance(other, Resource):
             return super().__eq__(other)
         else:
             return self.resource_id == other.resource_id and self.hostname == other.hostname \
                    and self.availability == other.availability and self.state == other.state \
                    and self.cpu_count == other.cpu_count and self.memory == other.memory \
-                   and self.total_cpu_count == other.total_cpu_count and self.total_memory == other.total_memory
-
-    def __init__(self, resource_id: str, hostname: str, availability: Union[str, ResourceAvailability],
-                 state: Union[str, ResourceState], cpu_count: int, memory: int, total_cpu_count: Optional[int],
-                 total_memory: Optional[int]):
-        super().__init__(pool_id=resource_id, hostname=hostname, cpu_count=cpu_count, memory=memory)
-
-        self._availability = None
-        self.availability = availability
-
-        self._state = state
-        self.state = state
-
-        self._total_cpu_count = cpu_count if total_cpu_count is None else total_cpu_count
-        self._total_memory = memory if total_memory is None else total_memory
+                   and self.total_cpus == other.total_cpus and self.total_memory == other.total_memory
+
+    def __init__(
+        self,
+        resource_id: str = None,
+        hostname: str = None,
+        availability: Union[str, ResourceAvailability] = None,
+        state: Union[str, ResourceState] = None,
+        cpu_count: int = None,
+        memory: int = None,
+        total_cpu_count: Optional[int] = None,
+        total_memory: Optional[int] = None,
+        **data
+        ):
+        if data:
+            # NOTE: this can be removed alias field names _are_ case sensitive
```

Comment:
Thanks for catching this, @christophertubbs! I need to double check this, but I _think_ this can be removed.

https://github.com/NOAA-OWP/DMOD/blob/62f7fbdfcad308d55d12bb5139d157b9b6225917/python/lib/scheduler/dmod/scheduler/resources/resource.py#L300-L302

Ill take a closer look when I revisit this. @aaraney #TODO